gmcs/tdl.py

```diff
@@ -706,28 +730,37 @@ def TDLmerge(e1, e2):
                 c0 += '\n\n'
             c0 += c2
         e0.set_comment(c0)
-
         # if the elements are ordered (list or dlist), merge the list
         # items in order.  That is, <a,b,c> + <A,B,C> = <a&A,b&B,c&C>.
+        # or (isinstance(e1,TDLelem_feat) and not e1.empty_list and e1.is_list())
         if e1.ordered():
             for i in range(max(len(e1.child), len(e2.child))):
                 if i < len(e1.child) and i < len(e2.child) and \
-                        TDLmergeable(e1.child[i], e2.child[i]):
+                        TDLmergeable(e1.child[i], e2.child[i]) and len(e2.child) > 0:
                     e0.add(TDLmerge(e1.child[i], e2.child[i]))
                 else:
-                    if i < len(e1.child):
+                    if i < len(e1.child) and len(e1.child) > 0:
                         e0.add(copy.copy(e1.child[i]))
-                    if i < len(e2.child):
+                    if i < len(e2.child) and len(e2.child) > 0:
                         e0.add(copy.copy(e2.child[i]))
```

Comment:
So there is more code around this which uses the same style (explicit call to len); do you think it should be part of the same PR to change all of that?

Reply:
It would be good to use a consistent style if this pattern is applicable to those cases. But since those changes wouldn't be related to replacing diff-lists with append-lists, they could also be relegated to some (near-)future PR. Maybe just make a new issue for code simplification/cleanup referring to one of these comments (hint: under the `...` menu of the comment, there's a "Reference in new issue" button).

Reply:
I created the issue.

Reply:
Ok that's fine. Let's fix them all up in another PR.